src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocMethodCheck.java

```diff
@@ -0,0 +1,357 @@
+ * }
+ * </pre>
+ * <ul>
+ * <li>
+ * Property {@code minLineCount} - Control the minimal amount of lines in method to allow no
+ * documentation. Default value is {@code -1}.
+ * </li>
+ * <li>
+ * Property {@code allowedAnnotations} - Configure the list of annotations that allow missed
+ * documentation. Default value is {@code Override}.
+ * </li>
+ * <li>
+ * Property {@code scope} - Specify the visibility scope where Javadoc comments are checked.
+ * Default value is {@code private}.
+ * </li>
+ * <li>
+ * Property {@code excludeScope} - Specify the visibility scope where Javadoc comments are
+ * not checked. Default value is {@code null}.
+ * </li>
+ * <li>
+ * Property {@code allowMissingPropertyJavadoc} - Control whether to allow missing Javadoc on
+ * accessor methods for properties (setters and getters). Default value is {@code false}.
+ * </li>
+ * <li>
+ * Property {@code ignoreMethodNamesRegex} - ignore method whose names are matching specified
+ * regex. Default value is {@code null}.
+ * </li>
+ * <li>
+ * Property {@code tokens} - tokens to check Default value is:
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#METHOD_DEF">METHOD_DEF</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#CTOR_DEF">CTOR_DEF</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ANNOTATION_FIELD_DEF">ANNOTATION_FIELD_DEF</a>.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the default check:
+ * </p>
+ * <pre>
+ * &lt;module name="MissingJavadocMethod"/&gt;
+ * </pre>
+ * <p>
+ * To configure the check for {@code public} scope:
+ * </p>
+ * <pre>
+ * &lt;module name="MissingJavadocMethod"&gt;
+ *   &lt;property name="scope" value="public"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * To configure the check for methods which are in {@code private}, but not in {@code protected}
+ * scope:
+ * </p>
+ * <pre>
+ * &lt;module name="MissingJavadocMethod"&gt;
+ *   &lt;property name="scope" value="private"/&gt;
+ *   &lt;property name="excludeScope" value="protected"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * To configure the check for ignoring methods named {@code foo(),foo1(),foo2()}, etc.:
+ * </p>
+ * <pre>
+ * &lt;module name="MissingJavadocMethod"&gt;
+ *   &lt;property name="ignoreMethodNamesRegex" value="^foo.*$"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * @since 8.20
+ */
+@FileStatefulCheck
+public class MissingJavadocMethodCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_JAVADOC_MISSING = "javadoc.missing";
+
+    /** Default value of minimal amount of lines in method to allow no documentation.*/
+    private static final int DEFAULT_MIN_LINE_COUNT = -1;
+
+    /** Specify the visibility scope where Javadoc comments are checked. */
+    private Scope scope = Scope.PRIVATE;
```

Comment:
@romani Do you want the scope to be PUBLIC just like the other check?

Reply:
yes, even scope type is very bad stuff to my mind.
Checks should be reasonable by default behavior, nobody writes javadoc in modern projects on non-API (public) methods.

Reply:
Done.